central/complianceoperator/v2/profiles/datastore/datastore.go

```diff
@@ -19,30 +20,38 @@ var (
 //go:generate mockgen-wrapper
 type DataStore interface {
 	// UpsertProfile adds the profile to the database
-	UpsertProfile(ctx context.Context, result *storage.ComplianceOperatorProfileV2) error
+	UpsertProfile(ctx context.Context, result *storage.ComplianceOperatorProfileV2, clusterID string, profileUID string) error
 
-	// DeleteProfile removes a profile from the database
-	DeleteProfile(ctx context.Context, id string) error
+	// DeleteProfileForCluster removes a profile from the database
+	DeleteProfileForCluster(ctx context.Context, uid string, clusterID string) error
+
+	// GetProfileEdgesByCluster gets the list of profile edges for a given cluster
+	GetProfileEdgesByCluster(ctx context.Context, clusterID string) ([]*storage.ComplianceOperatorProfileClusterEdge, error)
 }
 
 // New returns an instance of DataStore.
-func New(complianceProfileStorage pgStore.Store) DataStore {
+func New(complianceProfileStorage pgStore.Store, profileEdgeStore edge.Store, pool postgres.DB) DataStore {
 	ds := &datastoreImpl{
-		store: complianceProfileStorage,
+		store:            complianceProfileStorage,
+		profileEdgeStore: profileEdgeStore,
+		db:               pool,
 	}
 	return ds
 }
 
 // NewForTestOnly returns an instance of DataStore only for tests.
-func NewForTestOnly(_ *testing.T, complianceProfileStorage pgStore.Store) DataStore {
+func NewForTestOnly(_ *testing.T, complianceProfileStorage pgStore.Store, profileEdgeStore edge.Store, pool postgres.DB) DataStore {
 	ds := &datastoreImpl{
-		store: complianceProfileStorage,
+		store:            complianceProfileStorage,
+		profileEdgeStore: profileEdgeStore,
+		db:               pool,
 	}
 	return ds
 }
 
 // GetTestPostgresDataStore provides a datastore connected to postgres for testing purposes.
 func GetTestPostgresDataStore(_ *testing.T, pool postgres.DB) (DataStore, error) {
 	store := pgStore.New(pool)
-	return New(store), nil
+	edgeStore := edge.New(pool)
+	return New(store, edgeStore, pool), nil
```

Comment:
Should this be return NewForTestOnly(store, edgeStore, pool), nil

Reply:
Well actually `NewForTestOnly` should probably just be removed.

Reply:
Since I have another PR immediately after this, I'm going to remove it there.